src/cowrie/ssh_proxy/clientTransport.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2019 Guilherme Borges <guilhermerosasborges@gmail.com>
+# All rights reserved.
+
+from twisted.conch.ssh import transport
+from twisted.internet import defer, protocol
+from twisted.protocols.policies import TimeoutMixin
+from twisted.python import log
+
+from cowrie.core.config import CowrieConfig
+from cowrie.ssh_proxy.util import bin_string_to_hex, string_to_hex
+
+
+def get_int(data, length=4):
+    return int.from_bytes(data[:length], byteorder='big')
+
+
+def get_bool(data):
+    return bool(get_int(data, length=1))
+
+
+def get_string(data):
+    length = get_int(data, 4)
+    value = data[4:length+4]
+    return length+4, value
+
+
+class BackendSSHTransport(transport.SSHClientTransport, TimeoutMixin):
+    """
+    This class represents the transport layer from Cowrie's proxy to the backend SSH server. It is responsible for
+    authentication to that server, and sending messages it gets to the handler.
+    """
+    def __init__(self, factory):
+        self.delayedPackets = []
+        self.factory = factory
+        self.canAuth = False
+
+        # keep these from when frontend authenticates
+        self.frontendTriedUsername = None
+        self.frontendTriedPassword = None
+
+    def connectionMade(self):
+        log.msg('[CLIENT] New client connection from {0}'.format(self.transport.getPeer().host))
```

Comment:
can you add some more logging here, please? If this is the only logging message for a new connection, it needs similar structured information like is done for current ssh/telnet connections. (also a eventid so it gets picked up by the output modules)

Reply:
This is actually a connection to the backend, so I thought it wouldn't be too important since it should be the same for all attacker connections (which will change when the pool is added though). The frontend part has a more complete logging. Tell me what you think.

Reply:
Ah! I thought for a second it was front-end. then it's fine!